src/main/java/org/folio/auth/authtokenmodule/MainVerticle.java

```diff
@@ -456,316 +437,265 @@ private void handleAuthorize(RoutingContext ctx) {
-    if((tokenClaims.getBoolean("dummy") != null && tokenClaims.getBoolean("dummy"))
-            || username.startsWith(UNDEFINED_USER_NAME)) {
-      logger.debug("Using dummy permissions source");
-      usePermissionsSource = new DummyPermissionsSource();
-      dummyPermissionSource = true;
-    } else {
-      usePermissionsSource = permissionsSource;
-    }
+      PermissionsSource usePermissionsSource;
+      if (token.shouldUseDummyPermissionsSource()) {
+        logger.debug("Using dummy permissions source for token type: {}", token.getClaims().getString("type"));
+        usePermissionsSource = new DummyPermissionsSource();
+      } else {
+        usePermissionsSource = permissionsSource;
+      }
 
-    if (zapCache) {
-      usePermissionsSource.clearCache();
-    }
+      if (zapCache) {
+        usePermissionsSource.clearCache();
+      }
 
-    //Retrieve the user permissions and populate the permissions header
-    logger.debug("Getting user permissions for {} (userId {})", username, userId);
-    long startTime = System.currentTimeMillis();
+      //Retrieve the user permissions and populate the permissions header
+      logger.debug("Getting user permissions for {} (finalUserId {})", username, finalUserId);
+      long startTime = System.currentTimeMillis();
 
-    JsonArray expandedSystemPermissions = new JsonArray();
+      JsonArray expandedSystemPermissions = new JsonArray();
 
-    // Need to check if the user is still active
-    Future<Boolean> activeUser = Future.succeededFuture(Boolean.TRUE);
-    if (!dummyPermissionSource && finalUserId != null && !finalUserId.trim().isEmpty()) {
-      activeUser = userService.isActiveUser(finalUserId, tenant, okapiUrl, userRequestToken, requestId);
-    }
-    Future<PermissionData> retrievedPermissionsFuture = activeUser.compose(b -> {
-      if (b != null && b.booleanValue()) {
-        return permService.expandSystemPermissions(extraPermissions, tenant, okapiUrl, permissionsRequestToken,
-            requestId);
-      } else {
-        String msg = "Invalid token: user with id " + finalUserId + " is not active";
-        endText(ctx, 401, msg);
-        return Future.failedFuture(msg);
+      // Need to check if the user is still active.
+      Future<Boolean> activeUser = Future.succeededFuture(Boolean.TRUE);
+      if (token.shouldCheckIfUserIsActive(finalUserId)) {
+        activeUser = userService.isActiveUser(finalUserId, tenant, okapiUrl, userRequestToken, requestId);
       }
-    }).compose(expandedPermissions -> {
-      expandedSystemPermissions.addAll(expandedPermissions);
-      // skip expanded system permissions
-      JsonArray extraPermsMinusSystemOnes = new JsonArray();
-      extraPermissions.forEach(it -> {
-        if (!((String)it).startsWith(PermService.SYS_PERM_PREFIX)) {
-          extraPermsMinusSystemOnes.add(it);
+      Future<PermissionData> retrievedPermissionsFuture = activeUser.compose(b -> {
+        if (b != null && b.booleanValue()) {
```

Comment:
```suggestion
        if (Boolean.TRUE.equals(b)) {
```
And even
```
        if (TRUE.equals(b)) {
```
is possible after `import static java.lang.Boolean.TRUE;`

Reply:
Cool! Added it.